evaluator/src/main/kotlin/PackageRule.kt

```diff
@@ -84,15 +86,22 @@ open class PackageRule(
 
     /**
      * A [RuleMatcher] that checks whether any vulnerability for the [package][pkg] has a score that equals or is
-     * greater than [threshold] according to the [scoringSystem] and the belonging [severityComparator].
+     * greater than [threshold] according to the [scoringSystem] and the belonging [severityComparator]. Resolved
+     * vulnerabilities are ignored if a resolution exists in the [resolutionProvider].
      */
-    fun hasVulnerability(threshold: String, scoringSystem: String, severityComparator: (String, String) -> Boolean) =
+    fun hasVulnerability(
+        threshold: String,
+        scoringSystem: String,
+        resolutionProvider: ResolutionProvider = DefaultResolutionProvider.create(),
```

Comment:
Do we need a default value here? It introduces the risk that the parameter is forgotten.
If you add the `resolutionProvider` to the `PackageRule` as well in the previous commit, all matchers could access it without requiring an argument.

Reply:
The default is intended.
Using `DefaultResolutionProvider.create()` means that no resolution provider is being used and therefore no resolutions are applied.
I could also use `resolutionProvider: ResolutionProvider? = null` to make this more explicit.
What do you think?

Reply:
Added the `resolutionProvider` to the `PackageRule`.
However, I opted to still use a default:
```kt
fun packageRule(
        name: String,
        resolutionProvider: ResolutionProvider = DefaultResolutionProvider.create(),
        configure: PackageRule.() -> Unit
    )
```

Otherwise, this would break most existing `rules.kts`.